### PR TITLE
fix/type_label

### PR DESCRIPTION
**Pull request by Yongtae723**
I was checking to see how the learning is done with multilabel classification and then I found a type error.

I reported it and made a PR.

I think `y_label` can be List[List[int]] during multilabel classification.

I fixed some code that I found, but if you know some other place, let me know.
I will also fix and push that.

thank you!

### CODE REVIEW

**Review by tomaarsen — CHANGES_REQUESTED** [review]
Could you also change
https://github.com/huggingface/setfit/blob/90439c347054e9ba4c38cb710114b65e032b8249/src/setfit/data.py#L251
into
```python
    def __getitem__(self, idx: int) -> Tuple[TokenizerOutput, Union[int, List[int]]: 
```
Other than that, it looks good!

File: src/setfit/data.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ class SetFitDataset(TorchDataset):
     Args:
         x (`List[str]`):
             A list of input data as texts that will be fed into `SetFitModel`.
-        y (`List[int]`):
+        y (`List[int]` for single-label classification and `List[List[int]]` for multi-label classification):
             A list of input data's labels.
```
[the remark this answers is not in the document]
Comment (Yongtae723) [reply src/setfit/data.py]: I totally agree with you.

thank you for pointing that out.

**Comment by Yongtae723** [comment]
thank you for your kind review!

**Comment by tomaarsen** [comment]
Looks good to me! Thanks @Yongtae723

**Review by tomaarsen — APPROVED** [review]

**Pull request merged**